compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmBuiltinOptimizationLowering.kt

```diff
@@ -222,17 +225,34 @@ class JvmBuiltinOptimizationLowering(val context: JvmBackendContext) : FileLower
                 //      42.toLong()
                 //
                 // Doing so we avoid local loads and stores.
-                if (expression.statements.size == 2) {
-                    val first = expression.statements[0]
-                    val second = expression.statements[1]
+                if (statements.size == 2) {
+                    val first = statements[0]
+                    val second = statements[1]
                     if (first is IrVariable
                         && first.origin == IrDeclarationOrigin.IR_TEMPORARY_VARIABLE
                         && second is IrGetValue
                         && first.symbol == second.symbol) {
-                        expression.statements.clear()
-                        first.initializer?.let { expression.statements.add(it) }
+                        statements.clear()
+                        first.initializer?.let { statements.add(it) }
                     }
                 }
+            }
+
+            override fun visitBlockBody(body: IrBlockBody): IrBody {
+                body.transformChildrenVoid(this)
+                removeUnnecessaryTemporaryVariables(body.statements)
+                return body
+            }
+
+            override fun visitContainerExpression(expression: IrContainerExpression): IrExpression {
+                expression.transformChildrenVoid(this)
+                removeUnnecessaryTemporaryVariables(expression.statements)
+                return expression
+            }
+
+            override fun visitBlock(expression: IrBlock): IrExpression {
```

Comment:
Looks like this override is redundant, since super method delegates to `visitContainerExpression` which does the same?

Reply:
Good catch! Removed. :)